source/mir/ndslice/algorithm.d

```diff
@@ -1884,3 +1885,714 @@ size_t countImpl(alias fun, Slices...)(Slices slices)
     while(!slices[0].empty);
     return ret;
 }
+
+version(unittest)
+{
+    T identity(T)(T x)
```

Comment:
no need in global declaration. can be replaced with lambda

Reply:
I don't see how to do this with lambdas without making it more complicated. The lambdas I was trying weren't inferring the return type correctly, so then I tried making it a template and it just got uglier. I just put the identity function in each of the unittests instead of the version.